src/Reference_One.php

```diff
@@ -253,65 +119,32 @@ public function ref($defaults = [])
         });
 
         // if owner model is loaded, then try to load referenced model
-        if ($this->owner->loaded()) {
-            if ($this->their_field) {
-                if ($this->owner[$this->our_field]) {
-                    $m->tryLoadBy($this->their_field, $this->owner[$this->our_field]);
-                }
-
-                return
-                    $m->addHook('afterSave', function ($m) {
-                        $this->owner[$this->our_field] = $m[$this->their_field];
-                    });
-            } else {
-                if ($this->owner[$this->our_field]) {
-                    $m->tryLoad($this->owner[$this->our_field]);
-                }
-
-                return
-                    $m->addHook('afterSave', function ($m) {
-                        $this->owner[$this->our_field] = $m->id;
-                    });
+        if ($this->their_field) {
+            if ($this->owner[$this->our_field]) {
+                $m->tryLoadBy($this->their_field, $this->owner[$this->our_field]);
             }
-        }
 
-        // if owner model is not loaded, then return referenced model with condition set
-        // Imants: probably this piece of code should be moved to Reference_SQL_One->ref() method,
-        //         because only Persistence_SQL supports actions.
-        if (isset($this->owner->persistence) && $this->owner->persistence instanceof Persistence_SQL) {
-            $values = $this->owner->action('field', [$this->our_field]);
+            return
+                $m->addHook('afterSave', function ($m) {
+                    $this->owner[$this->our_field] = $m[$this->their_field];
+                });
+        } else {
+            if ($this->owner[$this->our_field]) {
+                $m->tryLoad($this->owner[$this->our_field]);
+            }
 
-            return $m->addCondition($this->their_field ?: $m->id_field, $values);
+            return
+                $m->addHook('afterSave', function ($m) {
+                    $this->owner[$this->our_field] = $m->id;
+                });
         }
 
         // can not load referenced model or set conditions on it, so we just return it
         return $m;
     }
 
-    // {{{ Debug Methods
-
     /**
-     * Returns array with useful debug info for var_dump.
-     *
-     * @return array
+     * List of properties to show in var_dump.
      */
-    public function __debugInfo()
-    {
-        $arr = [
-            'ref'     => $this->link,
-            'model'   => $this->model,
-        ];
-
-        foreach ([
-            'our_field', 'their_field', 'type', 'system', 'never_save', 'never_persist', 'read_only', 'ui', 'join',
-        ] as $key) {
-            if (isset($this->$key)) {
-                $arr[$key] = $this->$key;
-            }
-        }
-
-        return $arr;
-    }
-
-    // }}}
+    protected $debug_fields = ['our_field', 'their_field', 'type', 'system', 'never_save', 'never_persist', 'read_only', 'ui', 'join'];
```

Comment:
should be $__debug_fields instead of $debug_fields.
$link and $model should be added in $__debug_fields too if you agree with my other comment above in Reference::__debug_info method